### PR TITLE
Add and remove functionality

### PR DESCRIPTION
**Pull request by Saadat123456**
### Project requirements
- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for editing task descriptions.
- By default new tasks should have the property `completed` set to `false` and the property `index` set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
- All changes to the To-Do List should be saved in local storage.

### CODE REVIEW

File: src/js/todolist.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    this.list.forEach((task) => {
+      if (index !== task.index) {
+        newList.push(new Task(newIndex, task.completed, task.description));
+        newIndex += 1;
+      }
+    });
+    this.list = newList;
+    this.saveToLocalStorage();
+    TodoList.updateDOM(index);
+  }
```
Comment (oloomoses) [inline src/js/todolist.js:52]: - [ ] I have tried deleting a task and this action is not possible. Kindly implement it as per the project requirements.

File: src/js/todolist.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        listItem.setAttribute('id', `item${newIndex}`);
+        newIndex += 1;
+      }
+    });
+  }
+
+  editTaskDescription(index, description) {
+    this.list[index].description = description;
+    this.saveToLocalStorage();
+  }
```
Comment (oloomoses) [inline src/js/todolist.js:70]: - [ ] A task should be editable. Kindly ensure that this functionality is implemented as I was not able to edit a task

**Review by oloomoses — CHANGES_REQUESTED** [review]
Hi @Saadat123456, 

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/js/todolist.js (hunk 1 above)
Comment (Saadat123456) [reply src/js/todolist.js:52]: Hello @oloomoses !
Thanks for the review. Can you kindly check it again because I tested It and it is working.

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @Saadat123456 👋🏻, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💪🏻

✅ No linter errors.
✅ You used correct Github Flow.
✅ Professional README.
✅ Used webpack to bundle JavaScript.
✅ Implemented adding a new task
✅ Implemented editing task descriptions
✅ Implemented deleting a task


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**